poplar.mk

```diff
@@ -0,0 +1,304 @@
+	cp -f $(OPTEE_TEST_PATH)/out/xtest/xtest $(PKG_USR_BIN)
+	cp -f $(OPTEE_CLIENT_EXPORT)/lib/libtee* $(PKG_PATH)/usr/lib/aarch64-linux-gnu
+	find $(OPTEE_TEST_PATH)/out/ta -name "*.ta" -exec cp {} $(PKG_PATH)/lib/optee_armtz \;
+	echo "Package: op-tee" > $(PKG_CONTROL)
+	echo "Version: $(PKG_OPTEE_VERSION)" >> $(PKG_CONTROL)
+	echo "Section: base" >> $(PKG_CONTROL)
+	echo "Priority: optional" >> $(PKG_CONTROL)
+	echo "Architecture: arm64" >> $(PKG_CONTROL)
+	echo "Depends:" >> $(PKG_CONTROL)
+	echo "Maintainer: OP-TEE <op-tee@linaro.org>" >> $(PKG_CONTROL)
+	echo "Description: OP-TEE client binaries, test program and Trusted Applications" >> $(PKG_CONTROL)
+	echo " Package contains tee-supplicant, libtee.so, xtest and a set of" >> $(PKG_CONTROL)
+	echo " Trusted Applications." >> $(PKG_CONTROL)
+	echo " NOTE! This package should only be used for testing and development." >> $(PKG_CONTROL)
+	echo ""
+	dpkg-deb --build $(PKG_PATH)
+
+################################################################################
+# Prepare images
+################################################################################
+.PHONY: prepare-images
+prepare-images: linux l-loader rootfs
+	@cp $(TOOLS_PATH)/poplar_recovery_builder.sh $(OUT_PATH)
+	@cp $(LLOADER_BIN) $(OUT_PATH)
+	@cp $(LINUX_PATH)/arch/arm64/boot/Image $(OUT_PATH)
+	@cp $(LINUX_DTB) $(OUT_PATH)
+	@cd $(OUT_PATH) && \
+		PATH=$(UBOOT_PATH)/tools:$$PATH \
+	bash ./poplar_recovery_builder.sh all $(ROOTFS_BIN)
+
+################################################################################
+# Flash images
+################################################################################
+.PHONY: flash-help
+flash-help:
+	@echo "1. Install and configure TFTP server on your host PC:"
+	@echo ""
+	@echo "   $$ sudo apt-get install atftpd      # install atftpd server"
+	@echo "   $$ sudo vim /etc/default/atftpd     # edit atftpd server config"
+	@echo "   $$ sudo service atftpd restart      # restart atftpd server"
+	@echo ""
+	@echo "2. Proper configuration should look like:"
+	@echo ""
+	@echo "   $$ cat /etc/default/atftpd"
+	@echo "   USE_INETD=false"
+	@echo "   OPTIONS=\"--tftpd-timeout 300 --retry-timeout 5 --mcast-port 1758 --mcast-addr 239.239.239.0-255 --mcast-ttl 1 --maxthread 100 --verbose=5 /path/to/rep/out/dir\""
+	@echo ""
+	@echo "3. Flash proper U-boot build  to USB stick."
+	@echo "   It should be flashed to the first FAT32 partition. Then boot"
+	@echo "   from it by pressing USB_BOOT switch on the board"
+	@echo ""
+	@echo "4. Connect to Poplar board over serial console (run on host PC):"
+	@echo ""
+	@echo "   $$ screen /dev/ttyUSB0 115200"
+	@echo ""
+	@echo "5. Configure network interface in Poplar U-boot shell. If you can't"
+	@echo "get into U-boot console, press and hold Ctrl+C while booting:"
+	@echo ""
+	@echo "   => setenv ipaddr 192.168.0.2"
+	@echo "   => setenv netmask 255.255.255.0"
+	@echo "   => setenv serverip <your_host_ip_address>"
```

Comment:
Maybe helpful to mention that host ip is set to 192.168.0.3 as to example below.